src/Serialization/TypeValidatingEventSerializerTest.php

```diff
@@ -49,13 +43,8 @@ public function delegates_serialization_to_decorated_serializer()
      */
     public function delegates_unserialization_to_decorated_serializer()
     {
-        $event = new class() {};
-        $this->innerSerializer->method('unserializePayload')
-            ->with($this->equalTo(EventStub::class), $this->equalTo(['value' => 'some value']))
-            ->will($this->returnValue($event));
-
         $object = $this->serializer->unserializePayload(EventStub::class, ['value' => 'some value']);
-        $this->assertSame($event, $object);
+        $this->assertSame(['value' => 'some value'], $object->toPayload());
```

Comment:
If you want to decouple this from the underlying object you could do something like:

```php
$payloadArgs = [EventStub::class, ['value' => 'some value']]

$this->assertSame(
    $this->innerSerializer->unserializePayload(...$payloadArgs)->toPayload(),
    $this->serializer->unserializePayload(...$payloadArgs)->toPayload()
);
```

Reply:
Indeed, that'd be a good middle ground :smile: